subsys/bluetooth/mesh/settings.c

```diff
@@ -98,12 +104,10 @@ void bt_mesh_settings_store_schedule(enum bt_mesh_settings_flag flag)
 
 	if (atomic_get(pending_flags) & NO_WAIT_PENDING_BITS) {
 		timeout_ms = 0;
-	} else if (atomic_test_bit(pending_flags,
-				   BT_MESH_SETTINGS_RPL_PENDING) &&
-		   (!(atomic_get(bt_mesh.flags) & GENERIC_PENDING_BITS) ||
-		    (CONFIG_BT_MESH_RPL_STORE_TIMEOUT <
-		     CONFIG_BT_MESH_STORE_TIMEOUT))) {
-		timeout_ms = CONFIG_BT_MESH_RPL_STORE_TIMEOUT * MSEC_PER_SEC;
+	} else if (atomic_test_bit(pending_flags, BT_MESH_SETTINGS_RPL_PENDING) &&
```

Comment:
I think, this behavior together with the structure of `store_pending()` makes the RPL storage behavior dependent on other activity in the mesh stack. This does not align with what is mentioned in the documentation updates above. For example, RPL can still get frequently stored (even when set to MANUAL mode) after every CONFIG_BT_MESH_STORE_TIMEOUT after receiving any mesh message that changes any of the other persistent states. This breaks the behavior as one would expect by using long `RPL_STORE_TIMEOUT` or by using `BT_MESH_RPL_STORE_MANUAL.

Reply:
RPL won't be storing by any message is the pending flag won't be set. See:
https://github.com/zephyrproject-rtos/zephyr/blob/63cd8498ad28559ac7d3c51482132c3bce8386ee/subsys/bluetooth/mesh/rpl.c#L52-L54
https://github.com/zephyrproject-rtos/zephyr/blob/63cd8498ad28559ac7d3c51482132c3bce8386ee/subsys/bluetooth/mesh/rpl.c#L52-L54

Pending flag is only set when IV Update happens. See: https://github.com/zephyrproject-rtos/zephyr/blob/63cd8498ad28559ac7d3c51482132c3bce8386ee/subsys/bluetooth/mesh/rpl.c#L52-L54

`RPL_STORE_TIMEOUT` is set to (-1) which makes compiler entirely remove this `else-if`.